syntax = "proto3";

package p2;

service Serve {
    rpc RetrieveFile(ServeRequest) returns (ServeReply) {}
}

message ServeRequest {
    string session = 1;
    string url = 2;
    map<string, string> headers = 3;
}

message ServeReply {
    string session = 1;
    bool matching = 2;
    bytes data = 3;
    map<string, string> headers = 4;
}
